doc/source/release-notes/Percona-Server-5.7.21-21.rst

```diff
@@ -0,0 +1,135 @@
+  attempting to use partially-installed plugin. Bug fixed :psbug:`3959`.
+
+* There was a server crash caused by a materialized temporary table from
+  semi-join optimization with key length larger than 1000 bytes. Bug fixed
+  :psbug:`296`.
+
+* A regression in the original 5.7 port was causing integer overflow with
+  :variable:`thread_pool_stall_limit` variable values bigger than 2 seconds.
+  Bug fixed :psbug:`1095`.
+
+* A memory leak took place in |Percona Server| when performance schema is used
+  in conjunction with thread pooling. Bug fixed :psbug:`1096`.
+
+* A code clean-up was done to fix compilation with clang, both general warnings 
+  (bug fixed :psbug:`3814`, upstream :mysqlbug:`89646`) and clang 6 specific
+  warnings and errors (bug fixed :psbug:`3893`, upstream :mysqlbug:`90111`).
+
+* Compilation warning was fixed for `-DWITH_QUERY_RESPONSE_TIME=ON` CMake
+  compilation option, which makes QRT to be linked statically. Bug fixed
+  :psbug:`3841`.
+
+* |Percona Server| returned empty result for ``SELECT`` query if number of
+  connections exceeded 65535. Bug fixed :psbug:`314` (upstream
+  :mysqlbug:`89313`).
+
+* A clean-up in |Percona Server| binlog-related code was made to avoid
+  uninitialized memory comparison. Bug fixed :psbug:`3925` (upstream
+  :mysqlbug:`90238`).
+
+* ``mysqldump`` utility with ``--innodb-optimize-keys`` option was incorrectly
+  working with foreign keys on the same table, producing invalid SQL
+  statements. Bugs fixed :psbug:`1125` and :psbug:`3863`.
+
+* A fix of the mysqld startup script failed to detect jemalloc library
+  location for preloading, thus not starting on systemd based machines,
+  introduced in |Percona Server| :rn:`5.7.21-20`, was improved to take into
+  account previously created configuration file. Bug fixed :psbug:`3850`.
+
+* The possibility of a truncated bitmap file name was fixed in |InnoDB| logging
+  subsystem. Bug fixed :psbug:`3926`.
+
+* Temporary file I/O was not instrumented for Performance Schema. Bug fixed
+  :psbug:`3937` (upstream :mysqlbug:`90264`).
+
+* A crash in the unsafe query warning checks with views took place for
+  ``UPDATE`` statement in case of statement binlogging format. Bug fixed
+  :psbug:`290`.
+
+MyRocks Changes
+===============
+
+* A re-implemented variable :variable:`rpl_skip_tx_api` allows to turn on simple
+  RocksDB write batches functionality, increasing replication performance
+  by the transaction API skip. Bug fixed :jirabug:`MYR-47`.
+
+* Decoding value-less padded varchar fields could under some circumstances
+  cause assertion and/or data corruption. Bug fixed :jirabug:`MYR-232`.
+
+TokuDB Changes
+===============
+
+* Two new variables introduced for the |TokuDB| fast updates feature,
+  :variable:`tokudb_enable_fast_update` and
+  :variable:`tokudb_enable_fast_upsert` should be now used instead of the 
+  ``NOAR`` modifier, which is now optional at compile time and off by default.
```

Comment:
NOAR was never released in Percona Server in any way, and its all mentions should be removed. It was provided in Tokutek's server branches only